lib/utils/docker.coffee

```diff
@@ -281,29 +281,20 @@ exports.runBuild = (params, options, getBundleInfo, logStreams) ->
 
 			builder.createBuildStream(opts, hooks, reject)
 
-# Given an image id or tag, export the image to a tar archive.
-# Also needs the options generated by the appendOptions()
-# function, and a tmpFile to buffer the data into.
-exports.bufferImage = (docker, imageId, tmpFile) ->
+# Given an image id or tag, export the image to a tar archive,
+# gzip the result, and buffer it to disk.
+exports.bufferImage = (docker, imageId, bufferFile) ->
 	Promise = require('bluebird')
-	fs = require('fs')
-
-	stream = fs.createWriteStream(tmpFile)
+	streamUtils = require('./streams')
 
 	image = docker.getImage(imageId)
-	image.get()
-	.then (img) ->
-		new Promise (resolve, reject) ->
-			img
-			.on('error', reject)
-			.on('end', resolve)
-			.pipe(stream)
-	.then ->
-		new Promise (resolve, reject) ->
-			fs.createReadStream(tmpFile)
-			.on 'open', ->
-				resolve(this)
-			.on('error', reject)
+	imageMetadata = image.inspectAsync()
+
+	Promise.all([image.get(), imageMetadata.get('Size')])
```

Comment:
Conversely I find this is a little less clear - not by much mind you - but I would probably use a `.join` here.